fuse_core/include/fuse_core/eigen.h

```diff
@@ -79,6 +80,21 @@ std::string to_string(const Eigen::DenseBase<Derived>& m, const int precision =
   return oss.str();
 }
 
+template <typename Derived>
+bool isSymmetric(const Eigen::DenseBase<Derived>& m,
+                 const typename Eigen::DenseBase<Derived>::RealScalar precision =
+                     Eigen::NumTraits<typename Eigen::DenseBase<Derived>::Scalar>::dummy_precision())
+{
+  return m.isApprox(m.transpose(), precision);
+}
+
+template <typename Derived>
+bool isPSD(const Eigen::DenseBase<Derived>& m)
+{
+  Eigen::SelfAdjointEigenSolver<Derived> solver(m);
+  return solver.eigenvalues().minCoeff() > 0.0;
```

Comment:
With this we lose the ability to print the `eigenvalues()` when this check fails, as you can see in the code that uses `isPSD` in the rest of the changes in this PR.

If you think that information is important in case of error, we'd have to rethink this check. I think readability and encapsulation is more important in this case, since someone can also copy the matrix `m` into some software and compute the eigenvalues there or other stuff.

Reply:
I do like having this wrapped in a function. Having a simple function call makes it far more likely that I will test things that should be tested. And printing a message that says the matrix is not positive definite seems like sufficient information to track down the problem.